src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs

```diff
@@ -666,13 +680,19 @@ private async ValueTask<Stream> ConnectToTcpHostAsync(string host, int port, Htt
                     {
                         if (async)
                         {
-                            await socket.ConnectAsync(endPoint, cancellationToken).ConfigureAwait(false);
+                            IPAddress[] addresses = await Dns.GetHostAddressesAsync(host, cancellationToken).ConfigureAwait(false);
+                            activity?.AddEvent(new ActivityEvent("Dns.Resolved"));
```

Comment:
same here on events. Spans would provide much more value and flexibility than events.